qiime2/core/archive/provenance.py

```diff
@@ -373,8 +374,14 @@ def reference_plugin(self, plugin):
         return ForwardRef('environment:plugins:' + plugin.name)
 
     def capture_env(self):
-        return collections.OrderedDict(
-            (d.project_name, d.version) for d in pkg_resources.working_set)
+        env = collections.OrderedDict()
+
+        for pkg_list in importlib.metadata.packages_distributions().values():
```

Comment:
We learned this is incomplete, so I'm going to poke around a bit more for an equivalent to `working_set`